protocol/version.go

```diff
@@ -9,8 +9,10 @@ const (
 	Version36
 	Version37
 	Version38
+	Version39
 	VersionWhatever    VersionNumber = 0 // for when the version doesn't matter
 	VersionUnsupported VersionNumber = -1
+	VersionUnknown     VersionNumber = -2
```

Comment:
I'm not a big fan of having 3 sentinel values in this enum. I get what you use them for, but can we perhaps pass the information out of line (as a separate bool in the ParseHeader func), or just merge Unknown, Unsupported and Whatever?